src/org/jitsi/impl/neomedia/transform/rewriting/SsrcGroupRewriter.java

```diff
@@ -414,11 +415,25 @@ private void maybeSwitchActiveRewriter(final RawPacket pkt)
      */
     boolean isKeyFrame(RawPacket pkt)
     {
-        int sourceSSRC = pkt.getSSRC();
-        Byte redPT = ssrcRewritingEngine.ssrc2red.get(sourceSSRC);
-        byte vp8PT = 0x64;
-
-        return Utils.isKeyFrame(pkt, redPT, vp8PT);
+        byte pt = pkt.getPayloadType();
+        Byte redPT = null, codecPT = null;
+        String codecType = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
```

Comment:
I don't think I am following your concern here, since we don't have the codecType, we have to look it up anywhere we  are looking for keyframe, these two places don't share a common path to reuse the logic, does that make sense?